lib/util/source-code-fixer.js

```diff
@@ -108,7 +108,13 @@ SourceCodeFixer.applyFixes = function(sourceCode, messages) {
             // Make output to this fix.
             output += text.slice(Math.max(0, lastPos), Math.max(0, start));
             output += fix.text;
-            lastPos = end;
+
+            /*
+             * If the start of the range is larger than the end for some reason, make sure
+             * the text between the end and the start doesn't get duplicated.
+             * https://github.com/eslint/eslint/issues/8116
+             */
+            lastPos = Math.max(start, end);
```

Comment:
We could do that, but it might be a breaking change. Given that one of the core rules had a negative range, it seems likely that there are custom rules that use a negative range as well. The goal of this PR just is to fix the regression by restoring the previous behavior.